include/internal/param_names.h.in

```diff
@@ -0,0 +1,42 @@
+/*
+ * {- join("\n * ", @autowarntext) -}
+ *
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+{-
+use OpenSSL::paramnames qw(generate_internal_macros);
+-}
+
+typedef struct ossl_param_idx_s OSSL_PARAM_IDX;
+
+/* Parameter name definitions - generated by util/perl/OpenSSL/paramnames.pm */
+{- generate_internal_macros(); -}
+
+struct ossl_param_idx_s {
+    unsigned char p[NUM_PIDX];
+};
+
+int ossl_scan_params_for_idx(const OSSL_PARAM *params, OSSL_PARAM_IDX *indices);
+
+static ossl_inline OSSL_PARAM *ossl_param_idx(OSSL_PARAM *params,
+                                              const OSSL_PARAM_IDX *indices,
+                                              unsigned int n) {
+    fprintf(stderr, "looking for %d (%u)\n", n, indices->p[n]);
```

Comment:
Just for debug ?

Reply:
Yeah, this is really a work in progress to determine which approach we'll take for the final solution.